### ISSUE TITLE
"Missing" recipes for Thaumcraft vis crystals

### ISSUE THREAD

**Issue by Delfayne** [FTBTeam/FTB-Modpack-Issues#5684]
### Modpack

FTB Interactions Remastered

### Modpack version

1.0.1

### Has the pack been modified

No

### Describe the bug

Pressing R for recipes on Aer/Terra/Aqua/Ignis/Ordo/Entropy crystals do not show GT machine recipes.

However the recipes are configured, if you check Usages of e.g. aer-infused dust. This is potentially because the NBT of the two items is in the wrong order.

Thaumcraft's vis crystals have the NBT keys `aspect` then `amount`, whereas the recipes have `amount` then `aspect. This is confusing because FTBI 2.0 is consistent with the latter for both TC and the recipe. Or I might be going down the wrong path with thinking NBT is an issue.

### Steps to reproduce

Pressing R for recipes on Aer/Terra/Aqua/Ignis/Ordo/Entropy crystals do not show GT machine recipes.

### Expected behaviour

Recipes should display

### Screenshots

<img width="659" alt="image" src="https://github.com/user-attachments/assets/bbd5b697-582d-4e84-a678-3e8cd639fce3">

**Comment by ArtpokeBr** [comment]
Yeah it seems to be a NBT Issue, trying to fix it.

**Comment by ArtpokeBr** [comment]
Yeah, tested some things but not sure how to fix this one, JEI seems to always register things in specific way.

**Comment by Delfayne** [comment]
I recommend changing all references to vis crystals in scripts to use the ones in `_global.zs`, and swapping the `amount` and `key` tags in `_global.zs`. Ensure every reference consistently uses these keys in a strict order.

You can see this issue clearly play out in other ways. e.g. if you search for Ignis crystal in JEI and U it, there's no recipe involving an Astral altar for the ritual diviner, say. However if you go ignis dust then U, it shows the ignis crystal with a different NBT ordering THEN U that, you get to the diviner recipe.
